src/segments/mercurial.go

```diff
@@ -0,0 +1,166 @@
+package segments
+
+import (
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+)
+
+const (
+	MERCURIALCOMMAND = "hg"
+
+	hgLogTemplate = "{rev}|{node}|{branch}|{tags}|{bookmarks}"
+)
+
+type MercurialStatus struct {
+	ScmStatus
+}
+
+func (s *MercurialStatus) add(code string) {
+	switch code {
+	case "R", "!":
+		s.Deleted++
+	case "A":
+		s.Added++
+	case "?":
+		s.Untracked++
+	case "M":
+		s.Modified++
+	}
+}
+
+type Mercurial struct {
+	scm
+
+	Working           *MercurialStatus
+	IsTip             bool
+	LocalCommitNumber string
+	ChangeSetID       string
+	ChangeSetIDShort  string
+	Branch            string
+	Bookmarks         []string
+	Tags              []string
+}
+
+func (hg *Mercurial) Template() string {
+	return "hg {{.Branch}} ({{.LocalCommitNumber}}:{{.ChangeSetIDShort}}){{range .Bookmarks }} \uf02e {{.}}{{end}}{{range .Tags}} \uf02b {{.}}{{end}}{{if .Working.Changed}} \uf044 {{ .Working.String }}{{ end }}" //nolint: lll
+}
+
+func (hg *Mercurial) Enabled() bool {
+	if !hg.shouldDisplay() {
+		return false
+	}
+
+	displayStatus := hg.props.GetBool(FetchStatus, false)
+	if displayStatus {
+		hg.setMercurialStatus()
+	} else {
+		hg.Working = &MercurialStatus{}
```

Comment:
you could initialise this above and drop the initialisation at line 97. In that case you don't need and `if else`.